### PR TITLE
Add support for WPML (plugin / add-ons)

### PR DESCRIPTION
**Pull request by Spidlace**
## Checklist:
- [x] I've read the [Contributing page](https://github.com/junaidbhura/composer-wp-pro-plugins/blob/master/CONTRIBUTING.md).
- [x] I've created an issue and referenced it here.
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code has proper inline documentation.

## Description
Add support for WPML plugin and add-ons

### CODE REVIEW

File: plugins/WPML.php
Comment (MouseEatsCat) [inline plugins/WPML.php]: `in_array` should be replaced with `array_key_exists`

**Review by MouseEatsCat — CHANGES_REQUESTED** [review]

File: plugins/WPML.php
Comment (Spidlace) [reply plugins/WPML.php]: You're right, my bad! 
It's modified! :-)

**Review by MouseEatsCat — APPROVED** [review]
I tested this on a project with several WPML add-ons and it seems to work well!

File: plugins/WPML.php
Comment (mcaskill) [inline plugins/WPML.php:6]: Change file block comment to reflect WPML plugin.

```suggestion
/**
 * WPML Plugin.
 *
 * @package Junaidbhura\Composer\WPProPlugins\Plugins
 */
```

File: plugins/WPML.php
Comment (mcaskill) [inline plugins/WPML.php:13]: Change class block comment to reflect WPML plugin.

Rename the class and file name to CamelCase to reflect the other plugin class names.

```suggestion
/**
 * Wpml class.
 */
class Wpml {
```

File: plugins/WPML.php
Comment (mcaskill) [inline plugins/WPML.php:34]: Change method block comment to reflect WPML plugin.

```suggestion
    /**
     * Wpml constructor.
     *
     * @param string $version
     * @param string $slug
     */
```

File: plugins/WPML.php
Comment (mcaskill) [inline plugins/WPML.php:64]: Change to verbose array syntax and aligned array values to match coding style other classes.

```suggestion
        $packages = array(
            'wpml-sitepress-multilingual-cms'  => 6088,
            'wpml-string-translation'          => 6092,
            'wpml-media-translation'           => 7474,
            'wpml-woocommerce-multilingual'    => 637370,
            'wpml-gravityforms-multilingual'   => 8882,
            'wpml-contact-form-7-multilingual' => 3156699,
            'wpml-ninja-forms'                 => 5342487,
            'wpml-wpforms'                     => 5368995,
            'wpml-buddypress-multilingual'     => 2216259,
            'wpml-acfml'                       => 1097589,
            'wpml-all-import'                  => 720221,
            'wpml-mailchimp-for-wp'            => 1442229,
            'wpml-wp-seo-multilingual'         => 3566177,
            'wpml-types'                       => 1385906,
            'wpml-sticky-links'                => 6090,
            'wpml-cms-nav'                     => 6096,
            'wpml-translation-management'      => 6094,
        );
```

File: plugins/WPML.php
Comment (mcaskill) [inline plugins/WPML.php:68]: Added spacing to match coding style other classes.

```suggestion
        if ( array_key_exists( $this->slug, $packages ) ) {
            return 'https://wpml.org/?download=' . $packages[$this->slug] . '&user_id='. getenv( 'WPML_USER_ID' ) . '&subscription_key=' . getenv( 'WPML_KEY' ) . '&version=' . $this->version;
        }
```

**Review by mcaskill — CHANGES_REQUESTED** [review]
This feature is much appreciated.

I have not tested it myself yet but I propose a few minor changes to match the coding style of the rest of the package.

**Comment by junaidbhura** [comment]
Thanks everyone, happy new year!

Hey @Spidlace could you please implement the suggestions from @mcaskill ?

I'll be happy to merge it in once done. I'll make some minor updates once merged. I don't have an API key, @mcaskill have you verified this as working?

**Comment by mcaskill** [comment]
> […] I don't have an API key, @mcaskill have you verified this as working?

My colleague, @MouseEatsCat, and I [have tried it](https://github.com/junaidbhura/composer-wp-pro-plugins/pull/42#pullrequestreview-1225180307).

**Comment by junaidbhura** [comment]
Okay excellent @mcaskill - will wait for @Spidlace to implement the suggestions before merging it in

File: plugins/WPML.php
Comment (mcaskill) [inline plugins/WPML.php:35]: Fix the default slug to match the array key below.

```suggestion
    public function __construct( $version = '', $slug = 'wpml-sitepress-multilingual-cms' ) {
```

**Review by mcaskill — CHANGES_REQUESTED** [review]

File: README.md
Comment (mcaskill) [inline README.md:548]: This section is derived from from WP All Import/Export Pro and both should be rephrased to highlight the fact that only specific add-ons of each plugin are supported. The section should be phrase as so:

```suggestion
### WPML Add-Ons

The following WPML plugins and add-ons are supported:

* `junaidbhura/wpml-acfml`
* `junaidbhura/wpml-all-import`
* `junaidbhura/wpml-buddypress-multilingual`
* `junaidbhura/wpml-cms-nav`
* `junaidbhura/wpml-contact-form-7-multilingual`
* `junaidbhura/wpml-gravityforms-multilingual`
* `junaidbhura/wpml-mailchimp-for-wp`
* `junaidbhura/wpml-media-translation`
* `junaidbhura/wpml-ninja-forms`
* `junaidbhura/wpml-sitepress-multilingual-cms`
* `junaidbhura/wpml-sticky-links`
* `junaidbhura/wpml-string-translation`
* `junaidbhura/wpml-translation-management`
* `junaidbhura/wpml-types`
* `junaidbhura/wpml-woocommerce-multilingual`
* `junaidbhura/wpml-wp-seo-multilingual`
* `junaidbhura/wpml-wpforms`
```

**Review by mcaskill — CHANGES_REQUESTED** [review]
I noticed that the `Wpml.php` file is using spaces instead of tabs. That should also be fixed.

File: README.md
Comment (mcaskill) [inline README.md:445]: Fix JSON syntax.

```suggestion
  }
],
```

**Review by mcaskill — CHANGES_REQUESTED** [review]

**Comment by mcaskill** [comment]
Hi @Spidlace,

I hope you are doing well.

Will you be able to work on requested changes any time soon?
Alternatively, if your branch could add me as a contributor, I can make the changes for you.

Thanks and take care,

**Comment by mcaskill** [comment]
I've merged my suggestions in [my fork](https://github.com/mcaskill/composer-wp-pro-plugins/tree/feature/wpml), including some alleviation to the WPML code samples in the README.

I could create a separate pull request if we want to merge this feature sooner.

**Comment by junaidbhura** [comment]
@mcaskill Shall we close this PR in favour and maybe you could create a new PR instead?

**Comment by junaidbhura** [comment]
Closing in favour of https://github.com/junaidbhura/composer-wp-pro-plugins/pull/53

**Pull request closed without merging**
